### PR TITLE
update types to pass mypy --strict

### PR DESCRIPTION
**Pull request by nicolewhite**
if you run `mypy --strict` on the testing-sdk project in the examples repo you get a ton of errors, this fixes all of them

### CODE REVIEW

File: autoblocks/api/client.py [hunk 1]
```diff
@@ -1 +1,5 @@
-from autoblocks._impl.api.client import AutoblocksAPIClient  # noqa: F401
+from autoblocks._impl.api.client import AutoblocksAPIClient
+
+__all__ = [
```
Comment (nicolewhite) [inline autoblocks/api/client.py:3]: fixes https://mypy.readthedocs.io/en/stable/command_line.html#cmdoption-mypy-no-implicit-reexport

File: autoblocks/_impl/testing/models.py [hunk 2]
```diff
@@ -71,7 +79,7 @@ def hash(self) -> str:
         return self._cached_hash
 
 
-class BaseTestEvaluator(abc.ABC):
+class BaseTestEvaluator(abc.ABC, Generic[TestCaseType, OutputType]):
```
Comment (nicolewhite) [inline autoblocks/_impl/testing/models.py:82]: evaluators need generics for test case and output, so this now matches how it's set up in TS

File: tests/autoblocks/test_prompt_manager.py [hunk 3]
```diff
@@ -36,11 +41,11 @@ def my_template(
 
 class MyExecutionContext(
     PromptExecutionContext[
-        None,
```
Comment (nicolewhite) [inline tests/autoblocks/test_prompt_manager.py:39]: couldn't figure out how to do optional generics so just switched back to an empty pydantic class

**Review by adamnolte — APPROVED** [review]
Can we add a check here to try to catch these? Or i guess we could do that in the examples repo too?

File: tests/autoblocks/test_run_test_suite.py [hunk 4]
```diff
@@ -317,7 +317,7 @@ def test_error_in_evaluator(httpx_mock):
     def test_fn(test_case: MyTestCase):
         return test_case.input + "!"
 
-    class MySyncEvaluator(BaseTestEvaluator):
+    class MySyncEvaluator(BaseTestEvaluator[MyTestCase, str]):
```
Comment (nicolewhite) [inline tests/autoblocks/test_run_test_suite.py:320]: they'll need to specify generics for their evaluators like this in order to get their mypy checks to pass

**Comment by nicolewhite** [comment]
@adamnolte yeah looking at that now. this went undetected because we're not running on the `tests/` directory in this repo but i need to figure out how to do `--strict` but with some flags relaxed, like `--no-untyped-def`

**Comment by nicolewhite** [comment]
@adamnolte added here: https://github.com/autoblocksai/python-sdk/pull/161/commits/5beffe88632e049e2754fa248663e2d665d58337

**Review by dmorton2297 — APPROVED** [review]

**Pull request merged**